crates/stages/src/stages/index_storage_history.rs

```diff
@@ -80,12 +88,39 @@ impl<DB: Database> Stage<DB> for IndexStorageHistoryStage {
             return Ok(ExecOutput::done(input.checkpoint()))
         }
 
-        let (range, is_final_range) = input.next_block_range_with_threshold(self.commit_threshold);
+        let mut range = input.next_block_range();
+        let first_sync = input.checkpoint().block_number == 0;
 
-        let indices = provider.changed_storages_and_blocks_with_range(range.clone())?;
-        provider.insert_storage_history_index(indices)?;
+        // On first sync it's faster to rebuild the history from scratch.
```

Comment:
did you mean something like "faster to rebuild the history in one go"? so there's no batches of ranges, but from zero to end

Reply:
> because on first sync there's already no history, so it's from scratch by definition

there is/can be coming from genesis. there's even `insert_genesis_history` function